### PR TITLE
Fix order inconsistency between Hive Metastore partition information and actual partition values

### PR DESCRIPTION
**Pull request by XiaohuiShang**
Test plan - tests already exist.

== NO RELEASE NOTE ==

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/PartitionLoader.java [hunk 1]
```diff
@@ -35,14 +38,19 @@
...
             throws IOException;
 
-    public List<HivePartitionKey> getPartitionKeys(Table table, Optional<Partition> partition)
+    public List<HivePartitionKey> getPartitionKeys(Table table, Optional<Partition> partition, String partitionName)
     {
         if (!partition.isPresent()) {
             return ImmutableList.of();
         }
         ImmutableList.Builder<HivePartitionKey> partitionKeys = ImmutableList.builder();
+        // partition information provided by Hive Metastore is out of order
```
Comment (zhenxiao) [inline presto-hive/src/main/java/com/facebook/presto/hive/PartitionLoader.java:47]: partition information provided by Hive Metastore could be out of order

File: presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestHiveMetastoreUtil.java [hunk 2]
```diff
@@ -200,4 +201,15 @@ public void testReconstructPartitionSchema()
         assertThatThrownBy(() -> reconstructPartitionSchema(ImmutableList.of(c1), 2, ImmutableMap.of(0, c2)))
                 .isInstanceOf(IllegalArgumentException.class);
     }
+
+    @Test
+    public void testExtractPartitionValues()
+    {
+        String p1 = "date_str";
```
Comment (zhenxiao) [inline presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestHiveMetastoreUtil.java:208]: shall we have a concrete example?
e.g.
datePartition = "20201221"
countryPartition = "united_states"

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java [hunk 3]
```diff
@@ -470,31 +470,51 @@ public static void renameFile(FileSystem fileSystem, Path source, Path target)
...
         }
         checkArgument(!inKey, "Invalid partition spec: %s", partitionName);
         values.add(unescapePathName(partitionName.substring(valueStart, partitionName.length())));
+        keys.add(unescapePathName(partitionName.substring(keyStart, keyEnd)));
 
-        return values.build();
+        if (!partitionColumnNames.isPresent() || partitionColumnNames.get().size() == 1) {
+            return values.build();
+        }
+        else {
```
Comment (zhenxiao) [inline presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java:512]: else { is not needed

**Review by zhenxiao — CHANGES_REQUESTED** [review]
thank you @xiaohuishang 
mostly good. a few minor things

File: presto-hive/src/main/java/com/facebook/presto/hive/PartitionLoader.java (hunk 1 above)
Comment (xiaohuishang) [reply presto-hive/src/main/java/com/facebook/presto/hive/PartitionLoader.java:47]: Ok, let me modify it

File: presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestHiveMetastoreUtil.java (hunk 2 above)
Comment (xiaohuishang) [reply presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/TestHiveMetastoreUtil.java:208]: Ok, let me give a concrete example

File: presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java (hunk 3 above)
Comment (XiaohuiShang) [reply presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/MetastoreUtil.java]: Yep, it is indeed redundant.  I remove it

**Review by zhenxiao — APPROVED** [review]
looks good
@XiaohuiShang could you please squash the 2 commits into 1?

**Comment by XiaohuiShang** [comment]
> looks good
> @XiaohuiShang could you please squash the 2 commits into 1?

Ok, no problem

**Pull request merged**